docs/source/trainer/using_the_trainer.rst

```diff
@@ -0,0 +1,484 @@
+       loss = model.loss(outputs, batch)
+
+       loss.backward()
+       optimizer.step()
+
+   # eval loop
+   for batch in eval_dataloader:
+       outputs, targets = model.validate(batch)
+       metrics.update(outputs, target)
+
+For the actual code, see :meth:`.Trainer._train_batch_inner` and the :meth:`.Trainer.eval` methods.
+
+Quick Tour
+----------
+
+Below is a quick tour of various elements with code snippets for your reference. See the more
+detailed sections in the navigation menu.
+
+Events & State
+~~~~~~~~~~~~~~
+
+The core principle of the Composer trainer is to make it easy to inject
+custom logic to run at various points in the training loop. To do this,
+we have events that run before and after each of the lines above, e.g.
+
+.. code:: python
+
+   engine.run_event("before_forward")
+   outputs = model.forward(batch)
+   engine.run_event("after_forward")
+
+Algorithms and callbacks (see below) register themselves to run on one
+or more events.
+
+We also maintain a :class:`.State` which stores the trainer's state, such as
+the model, optimizers, dataloader, current batch, etc (see
+:class:`.State`). This allows algorithms to modify the state at the
+various events above.
+
+.. seealso::
+
+    :doc:`Events<trainer/events>` and :class:`.State`
+
+Algorithms
+~~~~~~~~~~
+
+The Composer trainer is designed to easily apply our library of
+algorithms to both train more efficiently and build better models. These
+can be enabled by passing the appropriate algorithm class to ``algorithms``
+argument.
+
+.. testcode::
+
+   from composer import Trainer
+   from composer.algorithms import LayerFreezing, Mixup
+
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='160ep',
+                     algorithms=[
+                         LayerFreezing(freeze_start=0.5, freeze_level=0.1),
+                         Mixup(alpha=0.1),
+                     ])
+
+   # the algorithms will automatically be applied during the appropriate
+   # points of the training loop
+   trainer.fit()
+
+We handle inserting those algorithms into the training loop and in the
+right order.
+
+.. seealso::
+
+    Our :doc:`Algorithms<algorithms>` guide, and the individual
+    :doc:`Method Cards </method_cards/methods_overview>` for each algorithm.
+
+
+Optimizers & Schedulers
+-----------------------
+
+You can easily specify which optimizer and learning rate scheduler to
+use during training. Composer provides a library of various optimizers
+and schedulers, but you can also include one of your own.
+
+.. code:: python
+
+   from composer import Trainer
+   from composer.models import ComposerResNet
+   from composer.optim.scheduler import cosine_annealing_scheduler
+   from torch.optim import SGD
+
+   model = ComposerResNet(model_name="resnet50", num_classes=1000)
+   optimizer = SGD(model.parameters(), lr=0.1)
```

Comment:
This example should be updated to use the new scheduling abstraction.

Reply:
I think showing that it supports the pytorch native schedulers is OK here.